tree/dataframe/test/dataframe_cloning.cxx

```diff
@@ -403,3 +403,66 @@ TEST(RDataFrameCloning, ChangeSpec)
       gSystem->Unlink(name.c_str());
    }
 }
+
+ROOT::RDF::Experimental::RResultMap<TH1D> dataframe_cloning_vary_with_filters_analysis(ROOT::RDF::RNode &df)
+{
+
+   auto df1 = df.Define("jet_pt", []() { return ROOT::RVecF{30, 30, 30, 10, 10, 10, 10}; });
+
+   auto df2 = df1.Vary("jet_pt",
+                       [](ULong64_t) {
+                          return ROOT::RVec<ROOT::RVecF>{{31, 31, 31, 31, 11, 11, 11}, {29, 29, 29, 29, 9, 9, 9}};
+                       },
+                       {"rdfentry_"}, {"pt_scale_up", "pt_res_up"});
+
+   auto df3 =
+      df2.Define("jet_pt_mask", [](const ROOT::RVecF &jet_pt) { return jet_pt > 25; }, {"jet_pt"})
+         .Filter([](const ROOT::RVecI &jet_pt_mask) { return ROOT::VecOps::Sum(jet_pt_mask) >= 4; }, {"jet_pt_mask"});
+
+   auto df4 = df3.Vary(
+      "weights",
+      [](const ROOT::RVecF &jet_pt, const ROOT::RVecI &jet_pt_mask) {
+         // The next line triggered an error due to a previous faulty implementation
+         // that was not connecting the cloned varied action to the correct upstream
+         // varied filters. In particular, this lambda should be run after the `Filter`
+         // above, which checks that the number of jets that passes the `jet_pt_mask`
+         // is at least 4.
+         auto test = ROOT::VecOps::Take(jet_pt[jet_pt_mask], 4);
+         return ROOT::RVecD{-1, 1};
+      },
+      {"jet_pt", "jet_pt_mask"}, 2);
+
+   auto h = df4.Histo1D<ROOT::RVecF, double>({"NAME", "TITLE", 10, -10, 10}, "jet_pt", "weights");
+
+   auto vars = ROOT::RDF::Experimental::VariationsFor(h);
+
+   return vars;
+}
+
+TEST(RDataFrameCloning, VaryWithFilters)
+{
+   ROOT::RDataFrame d{1};
+   auto df = d.Define("weights", []() -> double { return 0; });
+   ROOT::RDF::RNode df_rnode = ROOT::RDF::AsRNode(df);
```

Comment:
you should not need this line in C++, RInterface implicitly converts to RNode (the line itself in the test is not a problem, but I hope it's not actually needed otherwise we messed up at some point :D ).

the situation is different in Python, PyROOT is not able to do the conversion automatically for some reason (but it's a bug/limitation of PyROOT).

Reply:
This is a slight problem due to the signature of `TriggerRun(ROOT::RDF::RNode &df)`. I can't form an l-value reference automatically through the `RInterface::operator RNode`. I could maybe avoid using this function here and just iterate over the `RResultMaps`, what do you think?

Reply:
the signature of TriggerRun is "wrong", it should be `TriggerRun(ROOT::RDF::RNode df)`

Reply:
Yes, I see it now. Let me maybe open another PR for that

Reply:
This is now #13162